File: algorithm/杭电ACM/ambition0109/1032.java
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2011-04-12 20:01:35
////Compiler: Java
//////////////////////////////////////////////////////
////Problem ID: 1032
////Problem Title: 
////Run result: Accept
////Run time:93MS
////Run memory:6116KB
//////////////////System Comment End//////////////////
import java.util.Scanner;


public class Main {
	public static void main(String args[]){
		Scanner in=new Scanner(System.in);
		while(in.hasNext()){
			int l=in.nextInt();
			int r=in.nextInt();
			int ll=l,rr=r;
			if(ll>rr){
				int t=ll; ll=rr; rr=t;
			}
			int ans=0;
			for(int i=ll;i<=rr;i++){
				int cnt=1;
				for(int t=i;t!=1;cnt++){
					if((t&1)==1){
						t=t*3+1;
					}else{
						t=t/2;
					}
				}
				if(cnt>ans) ans=cnt;
			}
			System.out.println(l+" "+r+" "+ans);
		}
	}
}

File: algorithm/HDU-ACM/ambition0109/1032.cpp
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2010-06-27 17:38:40
////Compiler: Visual C++
//////////////////////////////////////////////////////
////Problem ID: 1032
////Problem Title: 
////Run result: Accept
////Run time:15MS
////Run memory:232KB
//////////////////System Comment End//////////////////
#include<iostream>
using namespace std;

void main()
{
    int m,n,mm,nn,i,j,t,q;
    while(scanf("%d%d",&m,&n)!=EOF)
    {
		mm=m>n?n:m; nn=m>n?m:n;
        for(i=mm,q=0;i<=nn;i++)
        {
            for(j=i,t=1;j>1;t++)
            {
                if(j%2==0)
                    j/=2;
                else
                    j=j*3+1;
            }
            if(t>q)
                q=t;
        }
        printf("%d %d %d\n",m,n,q);
    }
}